rdflib/plugins/stores/auditable.py

```diff
@@ -53,12 +53,7 @@ def add(self, triple, context, quoted=False):
         lock = destructiveOpLocks["add"]
         lock = lock if lock else threading.RLock()
         with lock:
-            context = (
-                context.__class__(self.store, context.identifier)
-                if context is not None
-                else None
-            )
-            ctxId = context.identifier if context is not None else None
+            ctxId = context if context is not None else None
```

Comment:
Happy to make this change to the original PR if it's generally agreed, otherwise - seeing as it was a rebase of a rebase, I left the original untouched.